examples/domainmodel/test/cross-refs.test.ts

```diff
@@ -4,12 +4,12 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import type { AstNode, LangiumDocument, ReferenceDescription } from 'langium';
-import type { Domainmodel } from '../src/language-server/generated/ast';
+import type { AstNode, LangiumDocument, ReferenceDescription } from '../../../packages/langium/lib/index.js';
```

Comment:
For some reason, inside of `test` folders, the TypeScript resolution does not work to `langium`:

![image](https://github.com/langium/langium/assets/4377073/71fd28bd-e9bf-4b95-b347-58e6313b42cf)

Reply:
vscode relies on `tsconfig.json` to configure itself and therefore it needs have both `src` and `test` with `noEmit=true` in this config. Another `tsconfig.src.json` must be introduced to drive the regular source build. 

Because it is easier to define then to explain, I created the branch: https://github.com/langium/langium/tree/kaisalmen/esm-test-langium-import
It demonstrates this. All import of `src` in tests have been adapted to use `langium` or `langium/test`.

Reply:
Thanks for the explanation! Sure, can you push your commit to this PR branch?